lib/internal/main/worker_thread.js

```diff
@@ -143,12 +147,22 @@ port.on('message', (message) => {
     port.postMessage({ type: UP_AND_RUNNING });
     if (doEval) {
       const { evalScript } = require('internal/process/execution');
-      evalScript('[worker eval]', filename);
+      const name = '[worker eval]';
+      // This is necessary for CJS module compilation.
+      // TODO: pass this with something really internal.
+      ObjectDefineProperty(process, '_eval', {
```

Comment:
Would a symbol be better/possible here?

Reply:
This is to match with what the CJS loader expects for the `--eval` CLI option, unfortunately